src/libs/actions/Report.js

```diff
@@ -777,46 +777,30 @@ function fetchAllReports(
-                // content and improve chat switching experience by only downloading content we don't have yet.
-                // This improves performance significantly when reconnecting by limiting API requests and unnecessary
-                // data processing by Onyx.
-                const reportIDsWithMissingActions = _.chain(returnedReports)
-                    .map(report => report.reportID)
-                    .filter(reportID => ReportActions.isReportMissingActions(reportID, lodashGet(allReports, [reportID, 'maxSequenceNumber'])))
-                    .value();
-
-                // Once we have the reports that are missing actions we will find the intersection between the most
-                // recently accessed reports and reports missing actions. Then we'll fetch the history for a small
-                // set to avoid making too many network requests at once.
-                const reportIDsToFetchActions = _.chain(ReportUtils.sortReportsByLastVisited(allReports))
-                    .map(report => report.reportID)
-                    .reverse()
-                    .intersection(reportIDsWithMissingActions)
-                    .slice(0, 10)
-                    .value();
-
-                if (_.isEmpty(reportIDsToFetchActions)) {
-                    Log.info('[Report] Local reportActions up to date. Not fetching additional actions.');
-                    return;
-                }
+            // Let's get all the oldest possible actions from the top 10 reports and update them enough for the user
+            // to have to scroll to see/load more actions
+
+            // NOTE: At the moment I'm not doing the reportIDsWithMissingActions as I'm not sure if it's needed.
+            const reportIDsToFetchActions = _.chain(ReportUtils.sortReportsByLastVisited(allReports))
+                .map(report => report.reportID)
+                .reverse()
+                .slice(0, 10)
+                .value();
+
+            if (_.isEmpty(reportIDsToFetchActions)) {
+                Log.info('[Report] Local reportActions up to date. Not fetching additional actions.');
+                return;
+            }
 
-                Log.info('[Report] Fetching reportActions for reportIDs: ', false, {
-                    reportIDs: reportIDsToFetchActions,
-                });
-                _.each(reportIDsToFetchActions, (reportID) => {
-                    const offset = ReportActions.dangerouslyGetReportActionsMaxSequenceNumber(reportID, false);
-                    fetchActions(reportID, offset);
-                });
+            Log.info('[Report] Fetching reportActions for reportIDs: ', false, {
+                reportIDs: reportIDsToFetchActions,
+            });
+
+            _.each(reportIDsToFetchActions, (reportID) => {
+                const offset = ReportActions.dangerouslyGetReportActionsMaxSequenceNumber(reportID, false);
+                readOldestAction(reportID, offset);
+            });
```

Comment:
Most of these changes need to be removed. I'll try to break down everything and explain why it's incorrect...

**Actions need to be tied to user actions**

- We should not be calling `readOldestAction()` as a direct replacement for `fetchActions()` 
- `readOldestAction()` would be called when we... read the oldest action and no other time
- Reading the oldest action means that we are scrolling up on the report and new chat messages are loading 


**So what "action" should we use for this existing code?**

- None. IIRC we decided to remove this.
- Why does it exist?
- It's an optimization (possibly not needed) used by the front end to preload a bunch of chat messages for various recent reports that we know are "missing actions" based on the locally stored actions we already have.
- Why are we using a timer? If we keep the code should it be removed?
- We run this code at a significant delay to avoid performance issues from loading too many actions at once. If we are not removing the code right now we should keep the timer.

**What should we do for now?**

I think just remove the code entirely is fine. If we want to "preload" actions we can think of a better design and have it hook into the big "App Start" event. Chatted about this for a bit with @luacmartins in [this doc section](https://docs.google.com/document/d/1Mc2fvigSqPnMa6yv6FQk7_opRpjK9kg0e2Gzz9QTXWM/edit#bookmark=id.j0m1o2aysusd).

Reply:
I agree with this, feel like once we finish the refactor, we can figure if this is still needed and if that is the case come up with a better way of doing this.

Is this one of the reasons we sometimes have a white loading page?

Reply:
Cool. Let's remove it and create a follow up issue to investigate.

I'm not sure about the white loading page and haven't experienced that myself - could be related or not.